airbnb/knowledge-repo
# Have force_reindex also update the git repository

Hey there,

I'm not 100% positive this should be merged with master as it pertains to a design choice rather than a missing feature, but here's a quick fix for those looking to easily refresh the knowledge repository after a post has been added (vs. having to restart the app or set up a cron job). 

It makes the force_reindex function, accessible through the route '/debug/force_reindex', not only reindex the db, but also update the local git repository. 

The caveat is that the app must be running with --debug enabled and the knowledge_branch must be set to 'master'.

**Test Plan:** 
1. Run a local instance of the knowledge_repo with runserver and --debug enabled pointing onto a local clone of a git repository 
2. Submit a new post to the git repository from another machine
3. Visit the localhost:7000/debug/force_reindex URL and see if the git repo was properly updated and the new post is now visible


**Also, I wanted to use this opportunity to let you know that we just forked your project internally over here at Criteo!** We've been facing the exact same issues you described in your [blog post](https://medium.com/airbnb-engineering/scaling-knowledge-at-airbnb-875d73eff091#.2u1me45uy), so this tool and its use case absolutely resonated with us. 

We're starting small, but if all goes well it should become a tool widely used by our analysts. And we'll be glad to share any useful new feature we may end up coming up with. 

**Thanks a lot guys!**

Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj
